.github/workflows/get-translations.yml

```diff
@@ -31,17 +37,42 @@ jobs:
       - name: Fetch latest dev
         run: git fetch origin dev
 
-      - name: Get translations
-        run: npx ts-node -O '{"module":"commonjs"}' ./src/scripts/crowdin/translations/getTranslations.ts
-        env:
-          CROWDIN_API_KEY: ${{ secrets.CROWDIN_API_KEY }}
-          CROWDIN_PROJECT_ID: ${{ secrets.CROWDIN_PROJECT_ID }}
-
       - name: Authenticate GitHub CLI
         run: |
           echo ${{ secrets.GITHUB_TOKEN }} | gh auth login --with-token
 
+      # Build translations
+      - name: Build Crowdin project
+        id: build-crowdin
+        run: |
+          OUTPUT=$(npx ts-node -O '{"module":"commonjs"}' ./src/scripts/crowdin/translations/triggerBuild.ts)
+          echo "$OUTPUT"
+        shell: bash
+
+      - name: Await latest build to finish
+        env:
+          BUILD_ID: ${{ steps.build-crowdin.outputs.buildId}}
+        run: |
+          OUTPUT=$(npx ts-node -O '{"module":"commonjs"}' ./src/scripts/crowdin/translations/awaitLatestBuild.ts)
+          echo "$OUTPUT"
+        shell: bash
+
+      - name: Check build success
+        run: |
+          if [ "${{ steps.build-crowdin.outputs.buildSuccess }}" = "false" ]; then
+            echo "Build timed out, exiting"
+            exit 1
+          fi
```

Comment:
If the `buildSuccess` from the prior step was `"false"` the we exit early due to timing out. `exit 1` will be treated as a failure, and the workflow will stop here.